compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -824,16 +824,16 @@ object SymDenotations {
       }
 
       /** Is protected access to target symbol permitted? */
-      def isProtectedAccessOK = {
-        def fail(str: => String): Boolean = {
-          if (whyNot != null) whyNot append str
+      def isProtectedAccessOK: Boolean =
+        def fail(str: => String): false =
+          if whyNot != null then whyNot.append(str)
           false
-        }
         val cls = owner.enclosingSubClass
-        if (!cls.exists)
+        if !cls.exists then
+          val encl = if !ctx.owner.isConstructor then ctx else ctx.outersIterator.dropWhile(_.owner.isConstructor).next()
```

Comment:
The code does not handle secondary constructors. Will something like the following work?

```suggestion
          val encl = if ctx.owner.isPrimaryConstructor then ctx.enclosingClass.owner.enclosingClass else ctx
```

Reply:
The expressions are the same, as the next outer context which is not owned by a ctor will be the correct enclosing class; also, both tests will discover an enclosing package symbol. I prefer your suggestion, which avoids an iterator -- although I almost talked myself out of it: the average view is that we seem to be skipping our class for the enclosing class; though from a context perspective, we're really just pushing out to the true enclosing class. (The context of expressions to super constructors is always confusing.) However, it seems `isConstructor` works here, also for secondaries; `outersIterator` would just plow through a couple more contexts in that case.

This change is a smallish tweak, but a good opportunity to take a look at scala 3 contexts...

Reply:
Sorry, I wasn't clear. I meant that in a secondary constructor, we might not want to go to the outer class --- as the current class itself could extend the related class.

Reply:
It's obvious when you put it that way.